src/drc/tests/test_partupload.py

```diff
@@ -0,0 +1,162 @@
+
+            response = self.client.put(
+                part_url,
+                {'inhoud': part_files.pop(0)},
+                format='multipart'
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+            part.refresh_from_db()
+
+            self.assertNotEqual(part.inhoud, '')
+            self.assertEqual(part.complete, True)
+
+    def _mark_complete(self):
+        complete_url = get_operation_url('enkelvoudiginformatieobject_complete', uuid=self.eio.uuid)
+
+        response = self.client.put(complete_url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        data = response.json()
+        self.canonical.refresh_from_db()
+        self.eio.refresh_from_db()
+        file_url = get_operation_url('enkelvoudiginformatieobject_download', uuid=self.eio.uuid)
+
+        self.assertEqual(self.canonical.parts.count(), 0)
+        self.assertNotEqual(self.eio.inhoud.path, '')
+        self.assertEqual(self.eio.inhoud.size, self.file_content.size)
+        self.assertEqual(data['inhoud'], f'http://testserver{file_url}?versie=1')
+
+    def _download_file(self):
+        file_url = get_operation_url('enkelvoudiginformatieobject_download', uuid=self.eio.uuid)
+
+        response = self.client.get(file_url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.content, b'filecontentstring')
+
+    def test_upload_file_full_process(self):
+        self._create_metadata()
+        self._lock_document()
+        self._upload_part_files()
+        self._mark_complete()
+        self._download_file()
+
+    def test_upload_file_wrong_size(self):
+        self._create_metadata()
+        self._lock_document()
+
+        # change file size for part file
+        part = self.parts[0]
+        part.chunk_size = part.chunk_size + 1
+        part.save()
+
+        # upload part file
+        part_url = get_operation_url('partupload_update', uuid=part.uuid)
+        part_file = split_file(self.file_content, settings.CHUNK_SIZE)[0]
+
+        response = self.client.put(
+            part_url,
+            {'inhoud': part_file},
+            format='multipart'
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        error = get_validation_errors(response, 'nonFieldErrors')
+        self.assertEqual(error['code'], 'file-size')
+
+    def test_upload_not_locked(self):
+        self._create_metadata()
+        self._upload_part_files()
+
+        # mark as complete
+        complete_url = get_operation_url('enkelvoudiginformatieobject_complete', uuid=self.eio.uuid)
+
+        response = self.client.put(complete_url)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        error = get_validation_errors(response, 'nonFieldErrors')
+        self.assertEqual(error['code'], 'unlocked')
```

Comment:
Can you add a few more tests?

1a. Upload to the same part URL twice
1b. Upload a correct and then an incorrect size to the same part URL
2. Call complete when the file is not uploaded completely.
3. Update a document with a different filesize
4. Update a document meta data, and performa a GET to see what it looks like (so without having uploaded the file parts).
5. Update a document meta data, and try to download the file (of which the new version was not uploaded yet)

Reply:
3. Is the case where you replace the document, where the new document has a different filesize than the original.

Reply:
case `test_update_eio_only_file_without_size`is added